nexus/src/db/datastore/silo_group.rs

```diff
@@ -118,10 +119,12 @@ impl DataStore {
             .actor_required()
             .internal_context("fetching current user's group memberships")?;
 
-        use db::schema::silo_group_membership::dsl;
-        dsl::silo_group_membership
-            .filter(dsl::silo_user_id.eq(actor.actor_id()))
-            .select(SiloGroupMembership::as_returning())
+        use db::schema::{silo_group as sg, silo_group_membership as sgm};
+        paginated(sg::dsl::silo_group, sg::id, pagparams)
+            .inner_join(sgm::table.on(sgm::silo_group_id.eq(sg::id)))
+            .filter(sgm::silo_user_id.eq(actor.actor_id()))
+            .filter(sg::time_deleted.is_null())
+            .select(SiloGroup::as_returning())
```

Comment:
I confirmed with `debug_query` that this produces this (I commented out the paginated part because I don't know how to unbox it):

```sql
SELECT silo_group.id,
       silo_group.time_created,
       silo_group.time_modified,
       silo_group.silo_id,
       silo_group.external_id
  FROM silo_group
       INNER JOIN silo_group_membership ON
			silo_group_membership.silo_group_id
			= silo_group.id
 WHERE silo_group_membership.silo_user_id = $1
   AND (silo_group.time_deleted IS NULL);
```